Accordion Block

asthabh23 opened the pull request:
Fix #60 

Test URLs:
- Before: https://main--delta--hlxsites.hlx.page/us/en/skymiles/how-to-earn-miles/airline-partners
- After: https://issue-60--delta--hlxsites.hlx.page/drafts/asthabharga/us/en/skymiles/how-to-earn-miles/airline-partners

ramboz commented on us/en/skymiles/blocks/accordion/accordion.js:
```diff
@@ -0,0 +1,100 @@
+
+  const parentWrapper = document.querySelectorAll('.accordion-wrapper');
+  parentWrapper.forEach((wrapper) => {
+    if (!wrapper.querySelector('.toolbar')) {
```
You can change this to a guard clause to save on the indentation:
```
if (wrapper.querySelector('.toolbar')) {
  return;
}

…
```

asthabh23 replied:
unable to follow this particular one

ramboz replied:
The idea of a _guard clause_ is just to reverse the conditional and exit early. that way you avoid indenting the main code and it usually makes it easier to read and maintain as the core of the method is essentially fully contained at the end in its own block.

It's nothing blocking the PR, but just a best practice that usually helps with long-term maintainability.

If you are not familiar with that pattern, here are a few quick examples:
- https://www.codementor.io/@clintwinter/use-guard-clauses-for-cleaner-code-1rrsczgwxp
- https://www.carloscaballero.io/refactoring-guard-clauses/

ramboz commented on us/en/skymiles/styles/styles.css:
```diff
@@ -21,6 +21,7 @@
   --color-delta-gray-light: #f6f6f7;
   --color-delta-red: #e51937;
   --color-delta-red-dark: #98012e;
+  --color-delta-border: #dedfe1;
```
You might be able to re-use `--color-delta-gray-middle` as it should be visually close

asthabh23 replied:
fixed in [f1d8070](https://github.com/hlxsites/delta/pull/89/commits/f1d80700d4d0f621c6466bffc47d9c5b6e4f461c)

ramboz commented on us/en/skymiles/styles/styles.css:
```diff
@@ -481,6 +482,10 @@ main img {
   }
+
+  main .section.accordion-container {
+    width: 100%;
```
Not sure this would have any effect since sections should already be displayed as block by default, and hence have 100% width

asthabh23 replied:
fixed in [f1d8070](https://github.com/hlxsites/delta/pull/89/commits/f1d80700d4d0f621c6466bffc47d9c5b6e4f461c)

ramboz commented on us/en/skymiles/scripts/scripts.js:
```diff
@@ -84,14 +84,14 @@ function decorateInlineToggles(container) {
-    el.replaceWith(span);
-  });
-}
+// function decorateScreenReaderOnly(container) {
```
I fixed the linting in main for this with an eslint disable. So you can probably just ignore this and revert the changes or rebase/merge the branch

asthabh23 replied:
yes, I rebased my branch, fixed in [f1d8070](https://github.com/hlxsites/delta/pull/89/commits/f1d80700d4d0f621c6466bffc47d9c5b6e4f461c)

ramboz commented on us/en/skymiles/blocks/accordion/accordion.css:
```diff
@@ -0,0 +1,135 @@
+}
+
+.bold {
+    font-weight: 1200;
```
Does this actually work? I thought 1000 was the max, and most fonts would max out at 700 and then not show a difference (unless they have a "black" variant which is bolder than "bold")

asthabh23 replied:
this was no longer required based on code optimisations, removed it from css file

ramboz reviewed (approved):
Looks good to me! Left a few comments for minor improvements

askayastha22 reviewed (changes requested):
@asthabh23 We need additional styling when an optional text is included within the image cell.
<img width="1126" alt="Screenshot 2023-03-09 at 12 07 35 PM" src="https://user-images.githubusercontent.com/108121450/224117227-80f0b9c7-9cd3-4fb0-b759-8e2fbfbc66a9.png">
<img width="1170" alt="Screenshot 2023-03-09 at 12 06 32 PM" src="https://user-images.githubusercontent.com/108121450/224117253-f5615976-57e5-4b36-9fd3-15886182f1a2.png">

askayastha22 commented:
@asthabh23 🐛 **Bug Alert**

https://user-images.githubusercontent.com/108121450/224128905-34d2536d-af07-4ae2-8c39-204868558b18.mov

asthabh23 commented:
@ramboz @askayastha22 Thanks for the feedbacks
I have incorporated them in my latest commit, fixed the bug and also added styling for the optional text.
Please have a look and merge into main if all fine :)

ramboz commented:
@askayastha22 Care to re-review now that @asthabh23 has addressed the remaining points?

askayastha22 commented:
@ramboz Yeah, I tested the new changes. Looks good. 👍
